### ISSUE TITLE
click.e.twitter.com (in Twitter marketing emails) inappropriately HTTPS-ified

### ISSUE THREAD

**Issue by wjt** [EFForg/https-everywhere#7137]
Twitter's marketing emails contain links to `http://click.e.twitter.com/`, which redirect to the actual target. For instance, an email about live streams of the US presidential debate has the following link: [Watch live](http://click.e.twitter.com/?qs=0f499d05ecc39ea79d921ad6b359dfa3c84362ec6616a75a44a5e8b004bfe47624131ec7a8ee1d59).

HTTPS Everywhere ships with a rule which matches this domain and `https`-ifies it, but the server serves a certificate only valid for `a248.e.akamai.net`, `*.akamaized.net`, `*.akamaihd-staging.net`, `*.akamaihd.net`, `*.akamaized-staging.net`.

(For what it's worth, `click.e.twitter.com` is a CNAME for `click.virt.s7.exacttarget.com`, a service run by Salesforce – which in turn is a CNAME for Akamai – rather than this being a tracking system run by Twitter themselves.)

**Comment by jeremyn** [comment]
Thanks for the report.

 [`Twitter.xml`](https://github.com/EFForg/https-everywhere/blob/90536e78269bf9b06bdd7c5e489cd47abd258640/src/chrome/content/rules/Twitter.xml) is the ruleset that matches `click.e.twitter.com`, specifically:

```xml
<target host="*.twitter.com" />
...
	<rule from="^http:"
		to="https:" />
```

 There are at least two possible fixes:

1. (Preferred but more work) Rewrite `Twitter.xml` to not use a wildcard target. Wildcards are [discouraged by the style guide](https://github.com/EFForg/https-everywhere/blob/34fb881e92ef2b9c871683da47093a49814dc2ec/ruleset-style.md), in part because they can break subdomains like we're seeing here.
2. (Less preferred, less work) Add `click\.e` to the [exclusion tag](https://github.com/EFForg/https-everywhere/blob/90536e78269bf9b06bdd7c5e489cd47abd258640/src/chrome/content/rules/Twitter.xml#L144) with an appropriate test.

**Comment by wjt** [comment]
@jeremyn I guess there's no analytics mechanism by which one could see which domains are, in practice, being matched by the wildcard?

**Comment by jeremyn** [comment]
Unfortunately, no. https://github.com/EFForg/https-everywhere/pull/6681#issuecomment-247343140 describes some approaches. For this specific issue though, for option (1), it would be enough to use the `test` URLs already in the ruleset.

**Comment by J0WI** [comment]
closed by #7416
